core/Pos/Arbitrary/Core.hs

```diff
@@ -169,6 +161,50 @@ instance HasCoreConstants => Arbitrary UnreasonableEoS where
               ]
     shrink = genericShrink
 
+----------------------------------------------------------------------------
+-- Address and related
+----------------------------------------------------------------------------
+
+instance Arbitrary Types.AddrType where
+    arbitrary =
+        oneof
+            [ pure Types.ATPubKey
+            , pure Types.ATScript
+            , pure Types.ATRedeem
+            , Types.ATUnknown <$> choose (3, maxBound)
+            ]
+
+instance Arbitrary Types.AddrSpendingData where
+    arbitrary =
+        oneof
+            [ Types.PubKeyASD <$> arbitrary
+            , Types.ScriptASD <$> arbitrary
+            , Types.RedeemASD <$> arbitrary
+            , Types.UnknownASD <$> choose (3, 255) <*> scale (min 150) arbitrary
```

Comment:
I just preserved the old logic. It means that the length of unparsed (unknown) data will be at most 150. I think it's ok. Probably it makes sense to decrease it further, but I don't think it's important.